core/controllers/payload_validator.py

```diff
@@ -81,10 +81,14 @@ def validate_arguments_against_schema(
     errors = []
     # Dictionary to hold normalized values of arguments after validation.
     normalized_values = {}
-
     for arg_key, arg_schema in handler_args_schemas.items():
-
-        if arg_key not in handler_args or handler_args[arg_key] is None:
+        if (
+                arg_key not in handler_args or
+                handler_args[arg_key] is None or
+                # Null request params are cast to strings by webapp.
```

Comment:
This actually seems off to me, so I want to dig into it.

In this case we are talking about JSON handlers, right? I thought what happens is that the frontend composes a dict, then that dict is encoded as JSON into a string, and that string is passed to the backend. The backend decodes the JSON to get the dict again and extracts the payload values.

In this flow I don't understand why null would be cast to a string since webapp wouldn't be looking at the value directly. Could you please clarify in what case this happens? The current approach feels like band-aiding the problem and I think it will introduce problems in the future since it introduces muddiness in the types -- we should fix the underlying problem of the "cast to string" if possible (which I thought was already handled by doing the JSON encoding).

Reply:
AFAICT, the JSON decoding only happens for the request _payload_ (core/controllers/base.py:174: `self.payload = json.loads(self.request.get('payload'))`), e.g. in POST requests, and not the request params, e.g. in DELETE requests which I believe webapp interprets as strings when it initializes the `request` object in core/controllers/base.py:164.

Thinking about it more, maybe the right thing to do here is to just omit the `language_code` request param in the request if `null` is supplied to the backend API.

Reply:
Got it, thanks. I think that makes sense for URL-based requests (i.e. your last paragraph) -- maybe leave a comment here to explain that as well.

That said -- @vojtechjelinek @Nik-09 could you weigh in on this please? (Since this affects our schema framework in general.) If we have any documentation about schemas we should update it to explain clearly how to deal with null parameter values in GET and DELETE requests.

Reply:
Thank you @seanlip & @sagangwee for mentioning this use case.

Yes we have wiki page for writing [schemas](https://github.com/oppia/oppia/wiki/Schemas) I will do a proofread and add essential points regarding this.
I will make changes and ask for a review by end of Sunday.

Thank you

Reply:
This change seems fine to me, but I think what we should also do is modify the `handler_args[arg_key]` on line 94 if it is a string and transform it to pure `None` there. So that the controller code receives pure `None` and not a string `'None'` or  `'null'`.

Reply:
@Nik-09 Any update on the wiki page?

Reply:
I have taken a look and realised that this change affects the architecture, not something which devs need to handle while writing schema.
(please correct me if I misinterpret)

Also @vojtechjelinek suggested some changes so it will be better to update wiki page (if required) once this PR is almost ready to merge.

Thank you

Reply:
@vojtechjelinek Thanks, I applied your suggestion and transformed the null string values.